teps/0049-aggregate-status-of-dag-tasks.md

```diff
@@ -117,6 +117,13 @@ will fail with the validation error if `$(tasks.status)` is used in any `dag` ta
 
 All necessary e2e, unit tests, and example will be added.
 
+## Alternatives
+
+Introduce the same variable but with an alternate name `$(pipeline.status)` instead of `$(tasks.status)`.
+
+The only downside to naming it this way is, it might confuse the users since this variable represents the overall status
+of the `tasks` section and not the entire `pipeline`.
```

Comment:
My intuition is the inverse since the status of the task dag execution ~= pipeline status in my mind but at the end of day, as long as it is documented, I think it is justifiable

Reply:
thanks @pierretasci, yup have documented in the implementation [PR](https://github.com/tektoncd/pipeline/pull/3817/).

> status of the task dag execution ~= pipeline status

yup its tricky, for a failure scenario, it is i.e. if one of the `dag` tasks fail, the `pipeline` status would be failure as well. But it is possible to have `dag` execution status to be successful while the `pipeline` status as failure if any of the finally task fail. The execution status of `finally` does impact the overall pipeline status.

Reply:
> status of the task dag execution ~= pipeline status

this is true before we execute `finally tasks`...it's not necessarily true after we run `finally tasks` -- so in this alternative, would we continue to `$(pipeline.status)` as we execute `finally tasks`?

it seems to me that `$(tasks.status)` would be clearer than `$(pipeline.status)`

Reply:
Both versions have pros and cons, the first feeling I get from `$(tasks.status)` is something that provides access to the status of various tasks. `$(pipeline.status)` gives more the idea of an aggregated status, but it's not entirely correct since it does not include the status of the finally part (obviously, since it's still running).
We could support both or have something more verbose, like `$(pipeline.before-finally.status)` which is really ugly...

Either case, nothing to hold this PR about :)